syntax = "proto3";
package solution;




import "app_store_sdk/model/app_store/micro_app.proto";

/*
ListSolutions请求
*/
message ListSolutionsRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页大小
    */
    int32 page_size = 2 ;
}

/*
ListSolutions返回
*/
message ListSolutionsResponse {
message List {
message Icon {
    /*
    图标库
    */
    string lib = 1 ;
    /*
    图标url
    */
    string icon = 2 ;
}
        
    
    /*
    关联的小产品
    */
    repeated app_store.AppStoreMicroApp microApps = 1 ;
    /*
    实例id
    */
    string instanceId = 2 ;
    /*
    名称
    */
    string name = 3 ;
    /*
    markdown格式的内容
    */
    string content = 4 ;
    /*
    轮播图片
    */
    string thumbnail = 5 ;
    /*
    简介
    */
    string brief = 6 ;
    /*
    图标
    */
    Icon icon = 7 ;
}
        
    
    /*
    返回总数
    */
    int32 total = 1 ;
    /*
    页数
    */
    int32 page = 2 ;
    /*
    该页大小
    */
    int32 page_size = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListSolutionsApi返回
*/
message ListSolutionsResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    solution.ListSolutionsResponse data = 4 ;
}

